### ISSUE TITLE
RFC: Math Prereqs are not recommended on the Discord

### ISSUE THREAD

**Issue by zachmmeyer** [ossu/computer-science#1211]
**Problem:**
Coming from someone who does not have a complete high school math education: When users mention they are following the [prereqs for math](https://github.com/ossu/computer-science/blob/master/FAQ.md#how-can-i-review-the-math-prerequisites) via Khan Academy on the Discord server they are given alternative paths to follow with reason to not use said resource. There are several pins in the prerequisite-math channel, going all the way back to what is currently recommended on this repo.

**Duration:**
1 month.

**Background:**
The two main recommendations I see are:

Lectures -- Professor Leonard (YouTube)
Homework/Assessments -- ALEKS, freely available from https://www.edx.org/course/precalculus
Extra Homework/Practice -- any suitable mass market college precalculus textbook available at your local Goodwill for 5 bucks, &/or OpenStax Precalculus
Additional help / tutoring -- Khan Academy

and

(These are also possibly paired with videos)
Pre-algebra: https://assets.openstax.org/oscms-prodcms/media/documents/Prealgebra2e-WEB_0qbw93r.pdf
Elementary Algebra: https://assets.openstax.org/oscms-prodcms/media/documents/ElementaryAlgebra2e-WEB_3zxfu3Z.pdf
Intermediate Algebra: https://assets.openstax.org/oscms-prodcms/media/documents/IntermediateAlgebra2e-WEB_RlpFLLx.pdf
Precalculus: https://assets.openstax.org/oscms-prodcms/media/documents/Precalculus_2e-WEB.pdf
High School Statistics: https://assets.openstax.org/oscms-prodcms/media/documents/Statistics-WEB.pdf

ALEKS is no longer available for free. We are recommended to not purchase it.
It was also not recommended to read the OpenStax books cover to cover, as it would be a waste of time. I do not know why.

There does not appear to be an agreed upon way forward. 
Both recommendations utilize OpenStax textbooks. 
As a newly non-free option ALEKS offers complimentary textbooks with their courses:
![image](https://github.com/ossu/computer-science/assets/34407561/77263017-3986-4285-89fc-2e45ca4fafea)

What was previously recommended with the links on the repo was to do the course assessments on Khan Academy to see how far you are along with your knowledge, but the issue is they only give you 30 questions per test and do not give a full assessment.

![image](https://github.com/ossu/computer-science/assets/34407561/af17c64b-e5fe-473e-9de7-290045ef3baa)

**Proposal:**
I do not know what the solution is here. I'd like the smart people who come up with these recommendations to have a concise, agreed upon solution that they could put on this repository for people to use so they do not have to feel like they need multiple sources of truth. @bradleygrant @pulkitkrishna00 I don't know who else I should tag in this.

**Alternatives:**
Keep things the way they are. Trying to complete the course challenges at Khan Academy one at a time until you know where you're at and then working through Integrated Math 1, 2, and 3 did not sound horrible to me until I shared that was what I was doing and then looked behind the curtain. It's what is recommended until it is not.

**Comment by zachmmeyer** [comment]
> The second "recommendation" mentioned is a quote from my message on discord. It is not supposed to be an _alternative_ to the first approach (which is from bradley). I was just listing the textbooks which you can use (the "first approach" mention using textbooks, I was just listing textbooks you can use which are legally available for free).
> 
> > ALEKS is no longer available for free. We are recommended to not purchase it.
> 
> You are not recommended to buy ALEKS from the ALEKS website, because you can get it cheaper from ASU. If you are willing to pay money, ALEKS is a very great tool.
> 
> > It was also not recommended to read the OpenStax books cover to cover, as it would be a waste of time. I do not know why.
> 
> Because there is quite an overlap between them.
> 
> > As a newly non-free option ALEKS offers complimentary textbooks with their courses:
> 
> You are welcome to use them. You can use _any_ textbook you want. For those who do not want to pay money. I listed the free openstax textbooks.

This is helpful clarification, thank you.

**Comment by bradleygrant** [comment]
This seems to be a request for an OSSU "prerequisite math academy".

For background: OSSU has long taken the approach that math prerequisites are "out of scope, done by others" because college CS majors begin in Calculus 1.

In truth, many college freshmen -- even college freshman entering a CS program -- need remedial math before starting Calculus 1. Many universities offer programs to get these students up to speed. (Perhaps not the likes of MIT or Stanford, but it's a very common situation at state universities.)

In principle, this represents a mission change for OSSU. Do we as an organization intend to sponsor and support prerequisite education for mathematics sufficient to allow people access to our programs, or do we intend to keep the status quo instead?

In practicality, we already have de facto recommendations on our Discord, and they run counter to the official recommendations on the OSSU curriculum. Should we *be offering* recommendations on the OSSU curriculum? Should we bring our recommendations into alignment?

**Comment by zachmmeyer** [comment]
Updated description to include limitations of Khan Academy.

**Comment by pulkitkrishna00** [comment]
To add to the discussion, the bioinformatics curriculum does have high school level math courses in it (although they are no longer active). Is there any discussion related to why they were added there?

---

We have 4 major sections in the CS curriculum:
- Prerequisites
- Intro CS
- Core CS
- Advanced CS

We *can* flesh out the prerequisite section (giving it equal treatment as the other sections, which is not the case for now) with prerequisite course recommendations. A note can be included indicating that prerequisite section can be skipped and is not a part of the curricular guidelines.

We can also move Py4E/CS50P to the "Prerequisite Programming" (along with the "Prerequisite Math" section under the "Prerequisites" heading), because they are also, in a practical sense, "prerequisites" and I don't think such courses are included in a typical CS curriculum.

If we go that road, I have an idea about including CS50 Scratch course (https://cs50.harvard.edu/scratch/2024/) under a "Prerequisite Programming".

But all that goes very much outside the scope of OSSU. The question from Bradley that "Do we as an organization intend to sponsor and support prerequisite education?" is still valid. I am not sure who can answer that.

**Comment by waciumawanjohi** [comment]
I've added a repository for a pre-college math repository and created a team to develop and maintain it. I've added those in this thread to that team; I'm happy to add other OSSU contributors to the team that have participated in developing another curriculum or advising learners on the discord.

Once the team feels the curriculum is ready for the light of day, it can be made public and PRs can be made in the other curricula to point to it for students in need of review.

**Comment by pulkitkrishna00** [comment]
This RFC has become struck as well. Developing a pre-college math has a question that needs to be answered first: Where do we set the base line? Algeba? Arithmetic? Counting? And if we are going that deep (very unlikely), we may as well cover English, and other sciences. I think it would be better if such discussions are done in public, with participation from students going through/starting/thinking to start prerequisite math.

I think there was an OSSU census held some times ago. Can we do something similar, trying to know what level of prerequisite Math would best serve OSSU students?

Also, I think for the time being, while this curriculum is being developed, the FAQ about prerequisite Math should be updated, to bring it in line with our recommendations on Discord (specifically, mention the "stack'). It is quite some confusion to many students.

**Comment by Flomza** [comment]
In my experience of teaching myself most of HS math through Khan Academy, then switching to "The Stack" for Precalculus as recommended on the Discord server, the best thing about Khan Academy is how streamlined it is. If you watch the videos and do the practice problems to 100% mastery, there's no jumping between resources and you just do it until it's done.

Now Professor Leonard has higher quality lectures than Khan Academy, and textbooks typically have higher quantity & quality practice problems than Khan Academy. For this, I agree that OSSU should move towards the recommendations found in the Discord pins.

If OSSU has an official "Prerequisite Math Academy" I feel like there should be some type of chart that maps Professor Leonard's lectures to associated practice problems in the textbook, sort of like a teacher assigning them as homework. If there is any way I can help contribute to this RFC, I would love to be a part of it :)

> Also, I think for the time being, while this curriculum is being developed, the FAQ about prerequisite Math should be updated, to bring it in line with our recommendations on Discord (specifically, mention the "stack').

Perhaps I could draft a PR to explain "The Stack" in FAQ?

**Comment by bradleygrant** [comment]
Sounds good. But don't offer to do it, do it.

**Comment by waciumawanjohi** [comment]
> If OSSU has an official "Prerequisite Math Academy" I feel like there should be some type of chart that maps Professor Leonard's lectures to associated practice problems in the textbook, sort of like a teacher assigning them as homework. If there is any way I can help contribute to this RFC, I would love to be a part of it :)

I had the same thought as Bradley. This sounds like a wonderful resource that doesn't exist yet in the world (I haven't seen it before, have you searched to ensure it doesn't already exist?). If you make it, it will be a benefit for all learners. And when you bring it to OSSU you can open an RFC to change the recommendation from Khan Academy to the new guide and explain why it's better.

([And remember to put it in front of users early and often while you are building.](https://www.gkogan.co/jeans/))
